paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -107,9 +133,103 @@ def paasta_mark_for_deployment(args):
     if service and service.startswith('services-'):
         service = service.split('services-', 1)[1]
     validate_service_name(service, soa_dir=args.soa_dir)
-    return mark_for_deployment(
+    ret = mark_for_deployment(
         git_url=args.git_url,
         deploy_group=deploy_group,
         service=service,
         commit=args.commit,
     )
+    if args.block:
+        try:
+            _log(
+                service=service,
+                line="Waiting for deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+            wait_for_deployment(args.service, args.deploy_group,
+                                args.commit, args.soa_dir, args.timeout)
+            _log(
+                service=service,
+                line="Deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+        except TimeoutError:
```

Comment:
Before exiting 1 I would like to make this as friendly as possible. Something like:

```
mark-for-deployment timed out after %s (human_time).

This probably means that the deploy hasn't succeeded. The new service might not be healthy or a cluster is having issues. Here is current status of the deploy right now:

print are_instance_deployed(for each cluster)

Things you can try to debug this:

  paasta status -vv -s $service
  paasta logs -s $service

$sha is still marked to be the desired deployment. If you would like to roll back, you can run:

    paasta rollback --service $service --deploy-group $deploy-group

```

Basically spoonfeed as much as we can to them for figuring out what this means, besides exit 1. We really can provide a lot without doing much more work.